test/Misc/stats_dir_long_path_name.swift

```diff
@@ -0,0 +1,16 @@
+// REQUIRES: OS=macosx
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: mkdir -p %t/very-long-directory-name-that-contains-over-128-characters-which-is-not-enough-to-be-illegal-on-its-own-but-presents
```

Comment:
I assume it doesn't matter that the total path at this point may or may not be over 256?

Reply:
nope. we're just trying to force it to be >256 when combined with the file and mangled into a string that has no path separators, so we need at _least_ two big segments in the combined path. Extra big segments in %t will just make it fail more-thoroughly.